File: backend/Jawlah.Core/Entities/Task.cs
using Jawlah.Core.Enums;

namespace Jawlah.Core.Entities;

public class Task
{
    public int TaskId { get; set; }
    public int AssignedToUserId { get; set; }
    public int? AssignedByUserId { get; set; }
    public int? ZoneId { get; set; }
    public string Title { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public TaskPriority Priority { get; set; }
    public Enums.TaskStatus Status { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime? DueDate { get; set; }
    public DateTime? StartedAt { get; set; }
    public DateTime? CompletedAt { get; set; }
    public double? Latitude { get; set; }
    public double? Longitude { get; set; }
    public string? LocationDescription { get; set; }
    public string? CompletionNotes { get; set; }
    public string? PhotoUrl { get; set; }
    public DateTime EventTime { get; set; }
    public DateTime? SyncTime { get; set; }
    public bool IsSynced { get; set; }
    public int SyncVersion { get; set; }
    public User AssignedToUser { get; set; } = null!;
    public User? AssignedByUser { get; set; }
    public Zone? Zone { get; set; }
}

File: web/src/types/task.ts
// Task types matching backend DTOs

export type TaskStatus = "Pending" | "InProgress" | "UnderReview" | "Completed" | "Rejected" | "Created" | "Assigned" | "Accepted" | "Submitted" | "Synced" | "Cancelled" | "FailedSync";
export type TaskPriority = "Low" | "Medium" | "High" | "Urgent";
export type TaskType = "GarbageCollection" | "StreetSweeping" | "ContainerMaintenance" | "RepairMaintenance" | "PublicSpaceCleaning" | "Inspection" | "Other";

export type TaskResponse = {
  taskId: number;
  sourceIssueId?: number;  // set if this task was created from an issue
  title: string;
  description: string;
  assignedToUserId: number;
  assignedToUserName: string;
  assignedByUserId?: number;
  assignedByUserName?: string;
  zoneId?: number;
  zoneName?: string;
  // Team task support
  teamId?: number;
  teamName?: string;
  isTeamTask: boolean;
  priority: TaskPriority;
  status: TaskStatus;
  taskType?: TaskType;
  requiresPhotoProof: boolean;
  estimatedDurationMinutes?: number;
  createdAt: string;
  scheduledAt?: string;
  dueDate?: string;
  startedAt?: string;
  completedAt?: string;   // server time (tamper-proof)
  eventTime?: string;     // device time when worker actually completed it
  locationDescription?: string;
  completionNotes?: string;
  photoUrl?: string;
  photos: string[];
  latitude?: number;
  longitude?: number;
  syncTime?: string;
  syncVersion: number;
  // Progress tracking for multi-day tasks
  progressPercentage: number;
  progressNotes?: string;
  extendedDeadline?: string;
  // Auto-rejection tracking
  isAutoRejected: boolean;
  rejectionReason?: string;
  rejectedAt?: string;
  rejectionDistanceMeters?: number;
  // Distance verification
  maxDistanceMeters: number;
  completionDistanceMeters?: number;
  isDistanceWarning: boolean;
};

export type CreateTaskRequest = {
  title: string;
  description: string;
  // For individual tasks - either assignedToUserId or teamId is required
  assignedToUserId?: number;
  // For team tasks - all team members can work on this task
  teamId?: number;
  zoneId?: number;
  priority: TaskPriority;
  taskType?: TaskType;
  requiresPhotoProof?: boolean;
  estimatedDurationMinutes?: number;
  scheduledAt?: string;
  dueDate?: string;
  latitude?: number;
  longitude?: number;
  locationDescription?: string;
};

export type UpdateTaskRequest = {
  title?: string;
  description?: string;
  assignedToUserId?: number;
  zoneId?: number;
  priority?: TaskPriority;
  taskType?: TaskType;
  requiresPhotoProof?: boolean;
  estimatedDurationMinutes?: number;
  scheduledAt?: string;
  dueDate?: string;
  latitude?: number;
  longitude?: number;
  locationDescription?: string;
};
